app/src/main/java/felixwiemuth/simplereminder/data/Reminder.java

```diff
@@ -78,17 +78,25 @@ public enum Status {
      */
     private final int naggingRepeatInterval;
     private @Setter String text;
+    private @Setter ReminderType reminderType;
     private @Setter Status status;
 
+    public enum ReminderType {
+        NORMAL,
+        NAGGING,
+        ALARM,
+    }
+
     @Builder //(builderClassName = "Builder")
-    public Reminder(int id, @NonNull Date date, int naggingRepeatInterval, @NonNull String text) {
+    public Reminder(int id, @NonNull Date date, int naggingRepeatInterval, @NonNull String text, @NonNull ReminderType reminderType) {
```

Comment:
Regarding changes to persistency (for example in this class), let's rather not continue on this for now because I haven't thought yet about how to do this best. (Several things have to be considered here including compatibility with the previous data format, whether it's "future save" and also that this class and others are now implemented in Kotlin.)